File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.js
/**
 * @param {string} colors
 * @return {boolean}
 */
var winnerOfGame = function(colors) {
//     one player remove a letter will not make chance for the other aaabbb=>aabb
//     look for aaa bbb pattern substr
    let AliceScore=0
    let BobScore=0
    for(let i=0;i<colors.length-2;i++){
        let subString=colors.slice(i,i+3)
        // console.log(subString)
        if(subString==="AAA") AliceScore++
        else if(subString==="BBB") BobScore++
    }
    // console.log(AliceScore,BobScore)
    return AliceScore>BobScore
};

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        sub_strs={"AAA":0,"BBB":0}
        for i in range(len(colors)-2):
            sub_str=colors[i:i+3]
            if sub_str=="AAA":
                sub_strs["AAA"]+=1
            
            if sub_str=="BBB":
                sub_strs["BBB"]+=1
        # print(sub_strs)     
        return sub_strs["AAA"]>sub_strs["BBB"]